lib/request.ex

```diff
@@ -1,4 +1,33 @@
 defmodule HTTPServer.Request do
-  defstruct method: :post, path: "", resource: nil, headers: %{}, body: ""
-  @type method :: :get | :post | nil
+  defstruct method: "", path: "", resource: "", headers: %{}, body: ""
+
+  @type t :: %__MODULE__{
+    method: String.t(),
+    path: String.t(),
+    resource: String.t(),
+    headers: %{optional(String.t()) => any},
+    body: String.t()
+  }
+
+  def parse_request(message) do
+    [request_data | body] = message |> String.split("\r\n\r\n")
```

Comment:
[Carriage return](https://github.com/kelseyroy/http_server/pull/6/files#r1220290460) here too 👀

Reply:
> For now I've added it as a module attribute to the modules that use it (Request, Response, and the HTTPServerTest)...

I mentioned this to Ian up above as well, I'm going to look into how I can make it even more global than this!